app/Http/Controllers/ProductsController.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Product;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Storage;
+use Illuminate\Support\Facades\Validator;
+
+class ProductsController extends Controller
+{
+    /**
+     * Get all employee data and display it
+     *
+     * @return \Illuminate\Contracts\Foundation\Application|\Illuminate\Contracts\View\Factory|\Illuminate\Contracts\View\View|\Illuminate\Foundation\Application
+     */
+    public function list()
+    {
+        return view('products.list', [
+            'products' => Product::all(),
+        ]);
+    }
+
+    /**
+     * Displays existing data into form
+     *
+     * @param Request $request
+     * @param Product $product
+     * @return \Illuminate\Contracts\Foundation\Application|\Illuminate\Contracts\View\Factory|\Illuminate\Contracts\View\View|\Illuminate\Foundation\Application
+     */
+    public function edit(Request $request, Product $product)
+    {
+        return view('products.edit', [
+            'filteredData' => $product,
+        ]);
+    }
+
+    /**
+     * Saves and validates new product entry. Also handles file upload
+     *
+     * @param Request $request
+     * @return \Illuminate\Contracts\Foundation\Application|\Illuminate\Foundation\Application|\Illuminate\Http\RedirectResponse|\Illuminate\Routing\Redirector
+     */
+    public function save(Request $request)
+    {
+        $validatedData = $request->validate([
+            'name' => ['required', 'max:60'],
+            'description' => ['required', 'max:255'],
+            'price' => ['required', 'numeric'],
+            'deliveryDate' => ['required', 'date'],
+            'productFile' => ['required', 'image', 'mimes:jpeg,png,jpg,gif', 'max:2048'],
+        ]);
+
+        $product = new Product();
+        $product->name = $validatedData['name'];
+        $product->description = $validatedData['description'];
+        $product->price = $validatedData['price'];
+        $product->date = $validatedData['deliveryDate'];
+        $product->fileType = $request->file('productFile')->getClientOriginalExtension();
+        $product->save();
+
+        if ($request->hasFile('productFile')) {
+            $file = $request->file('productFile');
+            Storage::put('public/productImages/' . $product->uuid, $file);
```

Comment:
Probably error? You should use `->move` method on file

Reply:
kliči move

Reply:
I will leave it as it is, because this solutions also works and I think it is readable.